### PR TITLE
Update Developer Guide Appendix

### LINKED ISSUE #271: Improve DG appendix

### PR DESCRIPTION
**Pull request by darkvoid32**
Closes #271.

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -663,16 +762,26 @@ testers are expected to do more *exploratory* testing.
...
 ### Saving data
 
 1. Dealing with missing/corrupted data files
 
-   1. _{explain how to simulate a missing/corrupted file, and the expected behavior}_
-</li>
+   1. Modify or delete data files
+
+   1. Open AssetBook. If AssetBook is open, close and reopen it.<br>
+      Expected: AssetBook will detect missing/corrupted data files, and start with an empty list
```
Comment (yisiox) [inline docs/DeveloperGuide.md:775]: Nit: Missing a period. Also I believe the present behavior shows a warning.

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -649,11 +737,22 @@ testers are expected to do more *exploratory* testing.
...
+
+1. Adding a contact
+
+    1. Test case: `add n\John Doe p\98765432 e\johnd@example.com a\311, Clementi Ave 2, #02-25 t\friends t\owesMoney A\screwdriver`<br>
+       Expected: AssetBook displays a success message, and the newly added contact's details are shown
+
+    1. Test case: `add n\John Doe`<br>
+       Expected: AssetBook displays an error stating invalid command format, and no contact is added
+
+    1. Other incorrect `add` commands to try: `add n\ p\ e\ a\311, Clementi Ave 2, #02-25 t\friends t\owesMoney A\screwdriver`, `...` <br>
```
Comment (yisiox) [inline docs/DeveloperGuide.md:750]: Is the `...` intended?

**Review by yisiox — COMMENTED** [review]
Some nits to check.

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -409,10 +409,12 @@ _{Explain here how the data archiving feature will be implemented}_
 
 **Target user profile**:
 
-* logistics managers who have a need to manage a significant number of contacts
-responsible for logistical assets
-* prefer desktop apps over other types
-* can type fast and prefers typing to other forms of input
+* logistics managers 
+* has a need to manage a significant number of contacts and their respective logistical assets
+* has a need to organise all assets
```
Comment (yisiox) [inline docs/DeveloperGuide.md:414]: Perhaps this two lines can merged to be
+ has a need to keep track of a significant number of contacts and assets they are responsible for

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (darkvoid32) [reply docs/DeveloperGuide.md]: Periods added for rest of test cases

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (darkvoid32) [reply docs/DeveloperGuide.md]: `...` removed

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (darkvoid32) [reply docs/DeveloperGuide.md]: merged statements

**Review by yisiox — APPROVED** [review]
LGTM

**Pull request merged**